extended-tracer/src/main/java/io/opentelemetry/contrib/tracer/Tracing.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.tracer;
+
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER;
+import static io.opentelemetry.api.trace.SpanKind.SERVER;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.baggage.BaggageBuilder;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+public final class Tracing {
+
+  private static final TextMapGetter<Map<String, String>> TEXT_MAP_GETTER =
+      new TextMapGetter<Map<String, String>>() {
+        @Override
+        public Set<String> keys(Map<String, String> carrier) {
+          return carrier.keySet();
+        }
+
+        @Override
+        @Nullable
+        public String get(@Nullable Map<String, String> carrier, String key) {
+          //noinspection ConstantConditions
+          return carrier == null ? null : carrier.get(key);
+        }
+      };
+  private final OpenTelemetry openTelemetry;
+
+  private final Tracer tracer;
+
+  public Tracing(OpenTelemetry openTelemetry, String tracerName) {
+    this.openTelemetry = openTelemetry;
+    this.tracer = openTelemetry.getTracer(tracerName);
+  }
+
+  public void run(String spanName, Runnable runnable) {
+    runAndEndSpan(tracer.spanBuilder(spanName).startSpan(), runnable);
+  }
+
```

Comment:
I think a useful extension for Runnables would be ability to wrap them in Spans, forming another Runnable which could be suitable for submission to Executor services for asynchronous execution.

Reply:
I agree, but would advocate for doing that in a follow-up PR.

Reply:
Relates to https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/9325